src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -415,12 +415,28 @@ public static void AppendKeyValue(this StringBuilder sb, string key, string valu
             if (includeQuotes)
             {
                 sb.Append('"');
+                int lastSpecialIndex = 0, specialIndex;
```

Comment:
Is this code performance critical? Simpler code would be value = value.Replace("\\", "\\\\").Replace("\"","\\\"");

Reply:
I didn't want to regress the performance since the value will almost never contain these characters.

Reply:
@davidsh, what are your thoughts? It's a trade off between readable code and performance so there's no "right" answer, which do you prefer in this scenario?

Reply:
Is it called per request? If yes, removing one allocation is good IMO.